### ISSUE TITLE
Milestone #4

### ISSUE THREAD

**Issue by eliseemelia** [geoffblay/DriftRacer#47]
Game Design Milestone #4
Final Project
 
1 Description

Milestone #4 is your final game deliverable. You will present your game and demonstrate it to me during the classes regularly scheduled final exam time and write a 15-page paper discussing your design choices and implementation issues.
2 Presentation

You are going to present your final game design and implementation to me during our regularly scheduled final exam time. You will have 20 minutes to do your presentation and game demonstration. I will enforce this deadline to ensure that all groups scheduled on that day have ample time to setup and present. Minimally, you should have:      

    A PowerPoint (or equivalent) slide presentation. Your presentation should last 10 minutes and contain the following:
        A description of your game:
            Overview (include genre, target audience), Story (background, setting, characters, in-game story), Rules (operational, constitutive, implicit, emergence, progression), goals, sub goals.
        An explanation of how you achieved descriptive and evaluative meaningful play in your game.
            You should discuss concepts from lecture, such as the core mechanic of your game, complexity, modes of interactivity, choices grouped by type, outcomes, juiciness (related to outcomes), uncertainty, goals and subgoals, feedback loops, conditioning (all types but especially rewards), flow, challenge, narrative, narrative descriptors, and your use of cutscenes, if any.  This is not an exhaustive list, and you should look back through all lectures.
        Discuss how your design and implementation changed through iterative development.  Start with a description of the evolution of the game due to the user testing conducted in Milestones 2 and 3.  Perform another round of user testing, like you did for the previous Milestones, and list how you changed your game one last time due to this new testing.
        (Remember to look at the presentation tips I gave you for Milestone 1.)
    A well-planned and comprehensive demonstration of your game. The game demonstration, live or recorded, must be kept under 10 minutes.  Remember, I'm looking for your ability to highlight game play and design, not technical implementation. Be sure to demonstrate "Haungs Mode".

3 Paper

You will write a detailed, 15-page, single spaced, 12pt font paper describing your game’s design and implementation. Your paper must include the following sections:

    Introduction
        Give a user-level description of your game. Include an overview of the story, the operational rules, game controls, and goal of the game.  This would be the type of description you would find in a game manual.
    Design
        Describe how your game achieves descriptive and evaluative meaningful play. You should discuss how you applied/used all of the concepts presented in lecture and how they affect meaningful play.  This is a more detailed and thorough description of what you presented in the above presentation.  In this section, your team is demonstrating mastery of game design concepts and principles.  THIS SECTION WILL DETERMINE A SIGNIFICANT PORTION OF YOUR MILESTONE 4 GRADE.
    Implementation
        Organization: At a high level, how are the components of your game interrelated. You should include a figure of your game’s organization.
        Game Objects: Discuss the major objects in your game and how they are implemented.
            C# Scripts:  Discuss the coding that you did to implement all of the major actions and behaviors in your game.
        Major Challenges: In the implementation of your game, you probably came across a few problems that were particularly challenging to design solutions for. Mention those here.
        Credit:  Please list the major online resources you used to help complete your game.  This would include coding systems you incorporated, models, UI, and any tutorials you referred to that had a major influence on the development of your game.
    Iterative Development
        Discuss the iterative development process you went through.  Summarize the iterations you already performed in Milestones 2 and 3.  Then, describe one full iteration you completed after Milestone 3.  Be sure to describe the changes you made after this last iteration.
            How did your game evolve?
            Be sure to include playtesting data in the form of graphs and/or tables for the new round of playtesting.
        Describe features that were dropped from your game due to playtesting.  This is very important!  As you tried to discover the best controls, goals, story, and challenges for your game through extensive playtesting, you undoubtedly came across portions of your implementation that had to be dropped or changed significantly.  This is where you get credit for all the work that did not make it into the final product.
    Publish
        Describe where you published your game and how I can access it to play it.
    Future Work (<1 page)
        What features do you think would be interesting to investigate adding to your game?
        If you had more time, what would you add to your game? More levels? Sub-goals?
    Conclusion
        A paragraph or two to reflect on the greatness of your game.

You will submit your paper before midnight on our regularly scheduled final exam day using Canvas.
4 Game

Your game must be 100% complete. At this stage, it would be better to drop a feature that is not entirely implemented, then to include it. Publish your game on Unity Play or itch.io.  You will provide me instructions on how to access your game in your paper.
Deliverables

You are responsible for the following:

    Presenting your game and giving a demonstration of it during our final exam time.
    Submitting your final paper (in pdf) on or before the midnight due date using Canvas.
